workspaces/src/cargo/mod.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::anyhow;
+use cargo_metadata::Message;
+use std::fmt::Debug;
+use std::path::Path;
+use std::process::{Command, Stdio};
+use tracing::debug;
+
+pub fn build_cargo_project<P: AsRef<Path> + Debug>(
+    project_path: P,
+) -> anyhow::Result<Vec<Message>> {
+    let cmd = Command::new("cargo")
```

Comment:
have you tried making use of the async version of `Command` (https://crates.io/crates/async-process)? That would probably be better to use since we're always gonna be in an async context. Starting up sandbox should switch over to that as well

Reply:
Done!